pubsub-adapter/src/main/kotlin/io/ably/lib/realtime/WrapperRealtimeClient.kt

```
@@ -0,0 +1,87 @@
+internal class WrapperRealtimeChannels(
+    private val javaChannels: AblyRealtime.Channels,
+    private val httpModule: Http,
+    private val agents: Map<String, String>,
+) :
+    Channels<RealtimeChannel> by RealtimeChannelsAdapter(javaChannels) {
+
+    override fun get(name: String): RealtimeChannel {
+        if (javaChannels.containsKey(name)) return WrapperRealtimeChannel(javaChannels.get(name), httpModule)
+        return try {
+            WrapperRealtimeChannel(javaChannels.get(name, ChannelOptions().injectAgents(agents)), httpModule)
+        } catch (e: AblyException) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)